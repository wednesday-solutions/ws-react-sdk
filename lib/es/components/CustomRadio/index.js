import{_ as r,a as o}from"../_rollupPluginBabelHelpers-c2a46d82.js";import"antd/lib/radio/style";import a from"antd/lib/radio";import"../_rollup-plugin-inject-process-env-1259e43d.js";import t from"react";import{c as n}from"../colors-93a064d8.js";import e from"styled-components";var i,l=e(a)(i||(i=r(["\n  .ant-radio .ant-radio-inner {\n    border-color: ",";\n  }\n  .ant-radio-checked .ant-radio-inner:after {\n    background-color: ",";\n  }\n"])),n.pawlyBlue,n.pawlyBlue);function d(r){return t.createElement(l,o({"data-testid":"custom-radio"},r))}export default d;
