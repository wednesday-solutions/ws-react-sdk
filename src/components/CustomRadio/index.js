/**
 *
 * CustomRadio
 *
 */

import React from 'react';
import { Radio } from 'antd';
import colors from '@themes/colors';
import styled from 'styled-components';

const CR = styled(Radio)`
  .ant-radio .ant-radio-inner {
    border-color: ${colors.pawlyBlue};
  }
  .ant-radio-checked .ant-radio-inner:after {
    background-color: ${colors.pawlyBlue};
  }
`;

function CustomRadio(props) {
  return <CR data-testid={'custom-radio'} {...props} />;
}

export default CustomRadio;
