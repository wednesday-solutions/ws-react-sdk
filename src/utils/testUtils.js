import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import { DEFAULT_LOCALE, translationMessages } from '@src/i18n';
import IntlGlobalProvider from '@components/IntlGlobalProvider';

export const renderWithIntl = children => {
  return render(
    <IntlProvider locale={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <IntlGlobalProvider>{children}</IntlGlobalProvider>
    </IntlProvider>
  );
};

export const getComponentStyles = (Component, props = {}) => {
  renderWithIntl(Component(props));
  const { styledComponentId } = Component(props).type;
  const componentRoots = document.getElementsByClassName(styledComponentId);
  // eslint-disable-next-line no-underscore-dangle
  return window.getComputedStyle(componentRoots[0])._values;
};

export const timeout = ms => new Promise(resolve => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});

export default {
  timeout,
  renderWithIntl,
  getComponentStyles,
  apiResponseGenerator
};
