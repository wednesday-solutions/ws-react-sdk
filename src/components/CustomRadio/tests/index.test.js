/**
 *
 * Tests for CustomRadio
 *
 */

import React from 'react';
import { renderWithIntl } from '@utils/testUtils';
import CustomRadio from '../index';

describe('<CustomRadio />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<CustomRadio />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 CustomRadio component', () => {
    const { getAllByTestId } = renderWithIntl(<CustomRadio />);
    expect(getAllByTestId('custom-radio').length).toBe(1);
  });
});
