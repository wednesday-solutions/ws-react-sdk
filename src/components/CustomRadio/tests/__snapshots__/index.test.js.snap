// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomRadio /> should render and match the snapshot 1`] = `
<body>
  <div>
    <label
      class="ant-radio-wrapper sc-bdfBwQ gaYMUQ"
    >
      <span
        class="ant-radio"
      >
        <input
          class="ant-radio-input"
          data-testid="custom-radio"
          type="radio"
          value=""
        />
        <span
          class="ant-radio-inner"
        />
      </span>
    </label>
  </div>
</body>
`;
